assembly/console.ts

```diff
@@ -0,0 +1,98 @@
+import { Promise } from "./Promise"
+
+declare function _log(data: string): void
+
+/* So Far, supports:
+- Strings
+- Numbers
+- UintArrays
+- IntArrays
+- Functions
+- ArrayBuffer
+- Map
+- Set
+- DataView
+- Promise
+- Array
+- StaticArray
+- Infinity
+Add more to the list if you think of any!
+*/
+
+export namespace console {
+    export function log<T>(data: T): void {
+
+        // -- String
+        if (data instanceof String) {
+            _log(changetype<string>(data))
+        }
+        // -- Number
+        else if (isFloat(data) || isInteger(data) || isSigned(data)) {
+            _log(f64(data).toString())
+        }
+        // -- Uint(8/16/32/64)Array
+        else if (data instanceof Uint8Array || data instanceof Uint16Array || data instanceof Uint32Array || data instanceof Uint64Array || data instanceof Uint8ClampedArray) {
+            _log(`UintArray(${data.length}) [${data.toString()}]`)
+        }
+        // -- Int(8/16/32/64)Array
+        else if (data instanceof Int8Array || data instanceof Int16Array || data instanceof Int32Array || data instanceof Int64Array) {
+            _log(`IntArray(${data.length}) [${data.toString()}]`)
+        }
+        // -- StaticArray
+        else if (data instanceof StaticArray) {
+            _log(data.toString())
+        }
+        // -- Array
+        else if (data instanceof Array) {
+            _log(data.toString())
+        }
+        // -- ArrayBuffer
+        else if (data instanceof ArrayBuffer) {
+            let array = Uint8Array.wrap(data)
+            let hex = ''
+            for (let i = 0; i < data.byteLength; i++) {
+                if (i === data.byteLength - 1) {
+                    hex += array[i].toString(16)
+                } else {
+                    hex += `${array[i].toString(16)} `
+                }
+            }
+            _log(`ArrayBuffer {\n  [Uint8Contents]: <${hex}>,\n  byteLength: ${data.byteLength}\n}`)
+        }
+        // -- Map
+        else if (data instanceof Map) {
+            _log('[object Map]')
```

Comment:
I suppose we'd want to improve this so that it logs something like

```
Map {
  [key1] => value1,
  [key2] => value2,
}
```

or similar. Otherwise logging a Map wouldn't be useful. But this is a good stub! 

I think it would be great to extract @jtenner's `Reflect` API (if he hasn't already?) into a separate lib, then that would allow us to easily log anything with any structure we want.

Reply:
Yeah, I have to work on that... :)

Reply:
He said the Reflect required a bunch of transforms and it was pretty slow

Reply:
I'm going to export this to a new repo...